clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -817,8 +821,12 @@ static void DiagnoseHLSLRegisterAttribute(Sema &S, SourceLocation &ArgLoc,
       S.Diag(TheDecl->getLocation(),
              diag::warn_hlsl_user_defined_type_missing_member)
           << regTypeNum;
-
-    return;
+    // non-zero SpaceNum cannot be specified for global constants
```

Comment:
This is validating that the space must be exactly `"0"`, whereas the rule that DXC implements is that you can't have any kind of space directive.